packages/dev/core/src/Meshes/Builders/shapeBuilder.ts

```diff
@@ -230,13 +250,56 @@ function _ExtrudeShapeGeneric(
         scaleFunction: Nullable<{ (i: number, distance: number): number }>,
         rotateFunction: Nullable<{ (i: number, distance: number): number }>,
         cap: number,
-        custom: boolean
+        custom: boolean,
+        adjustFrame: boolean
     ) => {
         const tangents = path3D.getTangents();
         const normals = path3D.getNormals();
         const binormals = path3D.getBinormals();
         const distances = path3D.getDistances();
-
+        if (adjustFrame) {
+            /* fix tangents,normals, binormals */
+            for (let i = 0; i < tangents.length; i++) {
+                if (tangents[i].x == 0 && tangents[i].y == 0 && tangents[i].z == 0) {
+                    tangents[i].x = tangents[i - 1].x;
+                    tangents[i].y = tangents[i - 1].y;
+                    tangents[i].z = tangents[i - 1].z;
+                }
+                if (normals[i].x == 0 && normals[i].y == 0 && normals[i].z == 0) {
+                    normals[i].x = normals[i - 1].x;
+                    normals[i].y = normals[i - 1].y;
+                    normals[i].z = normals[i - 1].z;
+                }
+                if (binormals[i].x == 0 && binormals[i].y == 0 && binormals[i].z == 0) {
+                    binormals[i].x = binormals[i - 1].x;
+                    binormals[i].y = binormals[i - 1].y;
+                    binormals[i].z = binormals[i - 1].z;
+                }
+                if (i > 0) {
+                    let v = tangents[i - 1];
+                    if (Vector3.Dot(v, tangents[i]) < 0) {
+                        //                        console.log(`fixing tangents[${i}]: ${tangents[i]}`);
+                        tangents[i].x *= -1;

```

Comment:
`.scaleInPlace(-1)`

Reply:
Should I push my changes that fix this?

Reply:
yes please, just push to the same branch and it will be updated here